components/polylith/parsing/core.py

```diff
@@ -12,13 +13,47 @@
     ".pytest_cache",
     "node_modules",
     ".git",
+    ".pixi",
 }
 
 
-def copy_tree(source: str, destination: str, patterns: set) -> Path:
-    ignore = shutil.ignore_patterns(*patterns)
+def is_match(root: Path, pattern: str, name: str, current: Path) -> bool:
+    path_name = (current / name).relative_to(root).as_posix()
 
-    res = shutil.copytree(source, destination, ignore=ignore, dirs_exist_ok=True)
+    return any(fnmatch.fnmatch(n, pattern) for n in [path_name, name])
+
+
+def any_match(root: Path, patterns: set, name: str, current: Path) -> bool:
+    return any(is_match(root, pattern, name, current) for pattern in patterns)
+
+
+def ignore_paths(root: Path, patterns: Set[str]):
+    def fn(current_path: str, names: List[str]):
+        current = Path(current_path).resolve()
+
+        return {name for name in names if any_match(root, patterns, name, current)}
+
+    return fn
```

Comment:
A custom ignore-function according to the way `shutil` uses `ignore`. The default `shutil.ignore_patterns` only handles names (files or folders) and will not ignore paths. That is why there's need of a custom ignore function.